### PR TITLE
updates for WebGLVectors

### PR DESCRIPTION
**Pull request by wendellwt**
My first attempt at adding support for WebGL Vectors

## Description

In src/components/sources copied `OlSourceWebglPoints.vue` to `OlSourceWebglVector.vue` and mostly changed Points to Vector,
In src/components/layers copied `OlWebglPointsLayer.vue` to `OlWebglVectorLayer.vue` and tried to change Points to Vector, and change it to Line in some places, but it should be 'Lines or Polygons', but wasn't sure how to do that, so it is just Line for now.

Also, since I don't know what I'm doing, I duplicated the `class WebGLVectorLayer extends Layer` in both files, which is most likely wrong.

## Motivation and Context

Desired support for WebGL Lines

## How Has This Been Tested?

I saw it work once!

### CODE REVIEW

File: src/components/sources/OlSourceWebglVector.vue [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  </div>
+</template>
+
+<script setup lang="ts">
+import VectorSource, { type Options } from "ol/source/Vector";
+import type { Ref } from "vue";
+import { inject, watch, onMounted, onUnmounted, provide, computed } from "vue";
+
+import Layer from "ol/layer/Layer.js";
+import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer.js";
```
Comment (d-koppenhagen) [inline src/components/sources/OlSourceWebglVector.vue:13]:
```suggestion
import Layer from "ol/layer/Layer";
import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer";
```

File: src/components/sources/OlSourceWebglVector.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  FEATURE_EVENTS,
+  useOpenLayersEvents,
+} from "@/composables/useOpenLayersEvents";
+
+// prevent warnings caused by event pass-through via useOpenLayersEvents composable
+defineOptions({
+  inheritAttrs: false,
+});
+
+const props = withDefaults(defineProps<Options<Line|Polygon>>(), {   // SHOULD prob. not be Line
```
Comment (d-koppenhagen) [inline src/components/sources/OlSourceWebglVector.vue:37]: It should / could be [`SimpleGeometry`](https://openlayers.org/en/latest/apidoc/module-ol_geom_SimpleGeometry-SimpleGeometry.html) to be more flexible right?

File: src/components/sources/OlSourceWebglVector.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  overlaps: true,
+  projection: "EPSG:3857",
+  useSpatialIndex: true,
+  wrapX: true,
+});
+
+const layer = inject<Ref<WebGLVectorLayer<VectorSource<Line|Polygon>>> | null>(   // SHOULD prob. not be Line
```
Comment (d-koppenhagen) [inline src/components/sources/OlSourceWebglVector.vue:44]: It should / could be [`SimpleGeometry`](https://openlayers.org/en/latest/apidoc/module-ol_geom_SimpleGeometry-SimpleGeometry.html) to be more flexible right?

File: docs/componentsguide/layers/webglvectorlayer/index.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# ol-webgl-vector-layer
+
+> A layer for rendering lines and polygons with WebGL
```
[the remark this answers is not in the document]
Comment (wendellwt) [reply docs/componentsguide/layers/webglvectorlayer/index.md:3]: I wasn't sure.  OpenLayers seemed to have them split into one with Points and another with Lines+Polygons, so I did the same.

File: src/components/sources/OlSourceWebglVector.vue (hunk 3 above)
Comment (wendellwt) [reply src/components/sources/OlSourceWebglVector.vue:44]: Does that mean that in `sources/OlSourceWebglVector.vue` that if it had:
```
import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer";

OlSourceWebglVector.vueclass WebGLVectorLayer extends Layer {
  createRenderer() {
    return new WebGLVectorLayerRenderer(this, {
      className: this.getClassName(),
      style: properties.style,
    });
  }
}
```
and:
```
const layer = inject<Ref<WebGLVectorLayer<VectorSource<SimpleGeometry>>> | null>
```
that it should work for all geom types?   I'll try doing that.
(That would mean that all of the current WebglPoints items would be taken over by this WebglVector one, right?)

Comment (d-koppenhagen) [reply src/components/sources/OlSourceWebglVector.vue]: I refactored it a little bit and extracted the class. Should work now.

**Comment by d-koppenhagen** [comment]
@wendellwt from my point of view, this is fine for now. Have you planned to change anything? Otherwise I can merge it and release a new version.

**Review by d-koppenhagen — APPROVED** [review]

**Comment by wendellwt** [comment]
I'm done with my updates, and have nothing else to add at this time.
I didn't delete the existing WebglPoint items, but since the new WebglVector works for points, you may want to remove the WebglPoint ones.
Thank you.

**Comment by d-koppenhagen** [comment]
That's fine. I will release it first as a feature version and later introduce the breaking change within the next major release

**Pull request merged**